packages/services/src/kernel/manager.ts

```diff
@@ -225,10 +229,24 @@ class KernelManager implements Kernel.IManager {
    * @returns A promise that resolves when all of the kernels are shut down.
    */
   shutdownAll(): Promise<void> {
+    // Proactively remove all models.
+    let previous = this._models.length;
+    if (previous) {
+      this._models = [];
+      this._runningChanged.emit([]);
+    }
+
     return this._refreshRunning().then(() => {
-      return Promise.all(this._models.map(model => this.shutdown(model.name))).then(() => {
-        return Promise.resolve(void 0);
-      });
+      return Promise.all(this._models.map(model => {
```

Comment:
How does this work if `this._models` has been set to `[]` above?

Reply:
Ha, whoops...